plugins/destination/duckdb/client/client.go

```diff
@@ -63,8 +63,10 @@ func New(ctx context.Context, logger zerolog.Logger, spec []byte, _ plugin.NewCl
 	}
 
 	// Unregister the built-in UUID type to avoid conflicts with our own UUID type
-	if err := arrow.UnregisterExtensionType(extensions.NewUUIDType().ExtensionName()); err != nil {
-		return nil, err
+	if _, ok := arrow.GetExtensionType(extensions.NewUUIDType().ExtensionName()).(*extensions.UUIDType); ok {
```

Comment:
Have 2 tests unregistering the type meant a test will not have `uuid` support. Added a check to ensure we only unregister the default one